geomstats/geometry/discrete_curves.py

```diff
@@ -440,3 +440,155 @@ def dist(self, point_a, point_b):
+                                 '2D Euclidean space.')
+
+        srv_metric = self.square_root_velocity_metric
+        srv = srv_metric.square_root_velocity(curve)[0]
+        srv_proj = srv_metric.project_srv(srv, atol=atol)
+        proj = srv_metric.square_root_velocity_inverse(srv_proj,
+                                                       gs.array([curve[0]]))
+
+        return proj
+
+
+class ClosedSRVMetric(SRVMetric):
+    """Elastic metric on closed curves.
+
+    See [Sea2011]_ for details.
+
+    Parameters
+    ----------
+    ambient_manifold : Manifold
+        Manifold in which curves take values.
+
+    References
+    ----------
+    .. [Sea2011] A. Srivastava, E. Klassen, S. H. Joshi and I. H. Jermyn,
+    "Shape Analysis of Elastic Curves in Euclidean Spaces,"
+    in IEEE Transactions on Pattern Analysis and Machine Intelligence,
+    vol. 33, no. 7, pp. 1415-1428, July 2011.
+    """
+
+    def __init__(self, ambient_manifold):
+        super(ClosedSRVMetric, self).__init__(ambient_manifold)
+
+    def project_srv(self, srv, atol=gs.atol):
+        """Project a point in the srv space into the space of closed curves srv.
+
+        The algorithm is from the paper cited above and modifies the srv
+        iteratively so that G(srv) = (0, ..., 0) with the paper's notation.
+
+        Parameters
+        ----------
+        srv : array-like, shape=[..., n_sampling_points, ambient_dim]
+            Discrete curve.
+        atol : float
+            Tolerance of the projection algorithm.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        proj : array-like, shape=[..., n_sampling_points, ambient_dim]
+        """
+        is_euclidean = isinstance(self.ambient_metric, EuclideanMetric)
+        is_planar = is_euclidean and self.ambient_metric.dim == 2
+
+        if not is_planar:
+            raise AssertionError('The projection is only implemented '
+                                 'for discrete curves embedded in a'
+                                 '2D Euclidean space.')
+
+        n_sampling_points = srv.shape[-2]
+        dim = self.ambient_metric.dim
+        srv_metric = self.l2_metric(n_sampling_points)
+        inner_prod = self.ambient_metric.inner_product
+
+        def g_criterion(srv, srv_norms):
+            return gs.sum(srv * srv_norms[:, None], axis=0)
+
+        initial_norm = srv_metric.norm(srv)
+        proj = srv
+        proj_norms = self.ambient_metric.norm(proj)
+        criteria = atol + 1
+        residual = g_criterion(proj, proj_norms)
+
+        while criteria >= atol:
+            g_jacobian = gs.zeros((dim, dim))
+            for i in range(dim):
+                for j in range(dim):
+                    g_jacobian[i, j] = 3 * inner_prod(proj[:, i], proj[:, j])
+            proj_squared_norm = srv_metric.squared_norm(proj)
+            g_jacobian += proj_squared_norm * gs.array([[1, 0], [0, 1]])
```

Comment:
Is the shape of this array linked to the dimension of the ambient space ? If it is, it would be nice to use ```dim``` to make it more general.